src/main/java/com/fzakaria/mvn2nix/maven/Maven.java

```diff
@@ -59,8 +79,62 @@ public Maven(File localRepository) {
 
         this.invoker.setOutputHandler(handler);
         this.invoker.setErrorHandler(handler);
+
+		File pomFile = new File("./pom.xml");
+		try {
+			this.pom = parseXml(pomFile);
+			this.repositories = getRepositoriesFromPom(pom);
+		}
+		catch (Exception e) {
+			throw new RuntimeException(e);
+		}
     }
 
+
+	/**
+	 * Parse an XML file
+	 * @param file
+	 * @return The parsed document
+	 * @throws ParserConfigurationException
+	 * @throws IOException
+	 * @throws SAXException
+	 */
+	public static Document parseXml(File file) throws ParserConfigurationException, IOException, SAXException {
+		DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+		DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+		Document doc = dBuilder.parse(file);
+
+		doc.getDocumentElement().normalize();
+		return doc;
+	}
+
+	/**
+	* Read repositories from the pom XML Document
+	* @param doc
+	* @return An ImmutableList of the found URLs
+	* @throws XPathException
+	* @throws MalformedURLException
+	 */
+	public static ImmutableList<URL> getRepositoriesFromPom(Document doc) throws XPathException, MalformedURLException {
+
+		XPath xPath = XPathFactory.newInstance().newXPath();
+		String expression = "/project/repositories/repository";
+
+		NodeList repositories = (NodeList) xPath.compile(expression).evaluate(doc, XPathConstants.NODESET);
+		LOGGER.debug("Found {} repositories in pom.xml", repositories.getLength());
```

Comment:
There are maven libraries for parsing the pom file.
I think they should be used instead ?

We also want to parse the ~/.m2/settings.xml file on the users machine.


```
ModelBuildingRequest req = new DefaultModelBuildingRequest();
req.setProcessPlugins( false );
req.setPomFile( file );
req.setModelResolver( new RepositoryModelResolver( basedir, pathTranslator ) );
req.setValidationLevel( ModelBuildingRequest.VALIDATION_LEVEL_MINIMAL );

Model model;
try
{
    model = builder.build( req ).getEffectiveModel();
}
catch ( ModelBuildingException e )
{
     ...
}
```

Reply:
Yes, that would be better. I don't have a lot of time to work on this anymore however.

Reply:
No worries; thank you for trying the contribution.
We can keep it open if others want to try and continue it or just close it then for now?

I'm also a bit swamped this quarter so I likely won't be able to make the addition myself.